fq_client.c

```diff
@@ -245,20 +245,47 @@ fq_resolve_endpoint(fq_conn_s *conn_s) {
     }
     memcpy(&conn_s->remote.sin_addr, *addr_list, sizeof(struct in_addr));
     free(buf);
+#elif defined _POSIX_C_SOURCE
+    struct addrinfo *result;
+
+    if (getaddrinfo(conn_s->host, NULL, NULL, &result)) {
```

Comment:
no mutex needed here?

Reply:
also, what about the first one?
`#ifdef HAVE_GETHOSTBYNAME_R`

Reply:
These alternative APIs are thread safe, they don't use static data like `gethostbyname` does